### ISSUE TITLE
FeatureRequest: Deep sorting

### ISSUE THREAD

**Issue by adamhamlin** [ronparkdev/eslint-plugin-sort-annotation#5]
This is a great plugin, but was wondering if it would be possible to allow for deep sorting of objects. So, for example, in the below object, the keys `a`, `b`, `c` have enforced sorting but `d`, `e`, `f` do not:

```ts
// @sort-keys
export const myObj = {
    a: '',
    b: '',
    c: {
        e: '',
        d: '',
        f: '',
    }
};
```

I can put another `// @sort-keys` annotation above `c` to enforce sorting on `d`, `e`, `f`, but that can become tedious/cumbersome for larger, highly-nested objects.

So maybe it would be something like `// @sort-keys:deep`?

**Comment by ronparkdev** [comment]
Hi @adamhamlin,

I felt the same need, but I wondered if anyone would need it. Thank you for your suggestion. I'll try to make it.

It's probably going to be `// @sort-keys:deep` as you say.

**Comment by ronparkdev** [comment]
@adamhamlin 

we're back in touch
We've implemented your feature and included it in version 1.0.6. 
You can try it out now.

As you suggested, you can use it like `// @sort-keys:deep`, 
or if you want to limit the depth, you can use it like `// @sort-keys:deep(2)`.

Please try it out and report any inconveniences or bugs.

Have a good day~!

**Comment by adamhamlin** [comment]
@ronparkdev thank you!
